scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -108,6 +108,13 @@ def add_parse_arguments(parser = None):
         help="Run only those tests that failed the previous twister run "
              "invocation.")
 
+    case_select.add_argument(
+        "--dyn-filter",
```

Comment:
need update to ducoment as well, and need add a simple command example on how to use it

Reply:
@hakehuang 
> need update to ducoment as well, and need add a simple command example on how to use it

You are right, needs more and better description for the user.

Idea is to reduce the CI load by only running tests on projects which have changed sources (compared to the git merge-base) and not every project.

At the moment the implementation needs to build all projects (to get the .ninja-deps file to make the association between changes sources and project) and then only executes the tests for those that have changed sources.

That only makes sense, where the build is much faster than the tests e.g. when simulating the MCU.
It also seems not complete. Any non-source project files such as prj.conf etc. are current not considered.

Furthermore the implementation could / should be improved by using the compile_commands.json (instead of ninja-deps).
- Improvement: no full build required as with .ninja-deps, compile_commands.json is created during cmake config phase
- Short-coming: (AFAIKT) there are build time generated sources which are not part of the compile_commands.json.
With using the compile_commands.json, which is created in the configuration phase of cmake, also the builds and not only the tests could be reduced.

For (long lasting) static code analysis (SCA) builds, when using ninja-deps, still a 1. stage build is required and a 2. stage build does the SCA. When using the compile_commands.json, the 1. stage build would not be necessary.

We tried to push the idea of that twister extension, but we also felt the reactions were rather reluctant.

We currently do not know whether or how we proceed.

Reply:
let's discuss this in test working group meeting, and feedback to you. we know this does not help the current CI, but using dependency is a good step forward. @romkell